pkg/console/controllers/oidcsetup/oidcsetup.go

```diff
@@ -0,0 +1,303 @@
+		crdLister:                 crdInformer.Lister(),
+		targetNSSecretsLister:     targetNSsecretsInformer.Lister(),
+		targetNSDeploymentsLister: targetNSDeploymentsInformer.Lister(),
+		targetNSConfigMapLister:   targetNSConfigMapInformer.Lister(),
+
+		authStatusHandler: status.NewAuthStatusHandler(authenticationClient, api.OpenShiftConsoleName, api.TargetNamespace, api.OpenShiftConsoleOperator),
+	}
+	return factory.New().
+		WithSync(c.sync).
+		ResyncEvery(wait.Jitter(time.Minute, 1.0)).
+		WithFilteredEventsInformers(
+			factory.NamesFilter("authentications.config.openshift.io"),
+			crdInformer.Informer(),
+		).
+		WithInformers(
+			authnInformer.Informer(),
+			consoleOperatorInformer.Informer(),
+			targetNSsecretsInformer.Informer(),
+			targetNSDeploymentsInformer.Informer(),
+			targetNSConfigMapInformer.Informer(),
+		).
+		ToController("OIDCSetupController", recorder.WithComponentSuffix("oidc-setup-controller"))
+}
+
+func (c *oidcSetupController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	if shouldSync, err := c.handleManaged(ctx); err != nil {
+		return err
+	} else if !shouldSync {
+		return nil
+	}
+
+	oidcClientsSchema, err := authnConfigHasOIDCFields(c.crdLister)
+	if err != nil {
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	// the schema is feature-gating this controller, we assume API validation won't
+	// allow authentication/cluster 'Type=OIDC' if the `.status.oidcClients` field
+	// does not exist
+	if !oidcClientsSchema {
+		// reset all conditions set by this controller
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("OIDCClientConfig", "", nil))
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("AuthStatusHandler", "", nil))
+		return statusHandler.FlushAndReturn(nil)
+	}
+
+	operatorConfig, err := c.consoleOperatorLister.Get(api.ConfigResourceName)
+	if err != nil {
+		return err
+	}
+
+	authnConfig, err := c.authnLister.Get(api.ConfigResourceName)
+	if err != nil {
+		return err
+	}
+
+	if authnConfig.Spec.Type != configv1.AuthenticationTypeOIDC {
+		applyErr := c.authStatusHandler.Apply(ctx, authnConfig)
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("AuthStatusHandler", "FailedApply", applyErr))
+
+		// reset the other condition set by this controller
+		statusHandler.AddConditions(status.HandleProgressingOrDegraded("OIDCClientConfig", "", nil))
+		return statusHandler.FlushAndReturn(applyErr)
+	}
+
+	// we need to keep track of errors during the sync so that we can requeue
+	// if any occur
+	var errs []error
+	syncErr := c.syncAuthTypeOIDC(ctx, syncCtx, statusHandler, operatorConfig, authnConfig)
+	statusHandler.AddConditions(
+		status.HandleProgressingOrDegraded(
+			"OIDCClientConfig", "OIDCConfigSyncFailed",
+			syncErr,
+		),
+	)
+	if syncErr != nil {
+		errs = append(errs, syncErr)
+	}
+
+	applyErr := c.authStatusHandler.Apply(ctx, authnConfig)
```

Comment:
if the syncError != nil, when we get here we won't have all the conditions specified.